pyproject.toml

```diff
@@ -0,0 +1,126 @@
+[project]
+name = "pygeoapi"
+version = "0.14.dev0"
+description = "pygeoapi provides an API to geospatial data"
+keywords = ["geospatial data api"]
+readme = "README.md"
+authors = [
+    {name = "Tom Kralidis", email = "tomkralidis@gmail.com"},
+]
+maintainers = [
+    {name = "Tom Kralidis", email = "tomkralidis@gmail.com"},
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Topic :: Scientific/Engineering :: GIS",
+]
+dependencies = [
+    "Babel",
+    "Flask",
+    "PyYAML",
+    "click>7,<=8",
+    "future-annotations",
+    "jinja2",
+    "jsonschema",
+    "pydantic",
+    "pygeofilter",
+    "pygeoif",
+    "pyproj",
+    "python-dateutil",
+    "pytz",
+    "rasterio",
+    "requests",
+    "shapely<2.0",
+    "tinydb",
+    "unicodecsv",
+    "Babel",
+    "click<=8,>7",
+    "Flask",
+    "future-annotations",
+    "jinja2==3.0.3",
+    "jsonschema",
+    "pydantic",
+    "pygeofilter",
+    "pygeoif==0.7",
+    "pyproj",
+    "python-dateutil",
+    "pytz",
+    "PyYAML",
+    "rasterio",
+    "requests",
+    "shapely<2.0",
+    "tinydb",
+    "unicodecsv",
+    "gdal==3.4.1",
+]
+license = {text = "MIT"}
+requires-python = ">=3.8"
```

Comment:
So we are sure that Python 3.10 and 3.11 will work? Maybe then also reflect in `main.yml` Python version matrix. Or be conservative `< 3.10`. Is 3.7 obsolete? For me hard to tell. I locally even run *Python 3.7.1 (default, Nov 17 2020, 13:42:41)* for pygeoapi with `pyenv` MacOSX.
The current latest Docker on demo runs:
*Python 3.8.10 (default, Nov 14 2022, 12:59:47)*

Reply:
Requiring 3.7.1 or greater would make dependency pinning stuck with some quite old versions, some of which are not even provided as a pre-compiled wheel. Python 3.7 will also reach end of life in a few months, so it shouldn't be a big issue, but I can extend compatibility to that too.

`main.yml` does not reflect that, since this PR still has some failing tests and is a work in progress, but I will take care of that too.

Reply:
@francbartoli I think that it would be the safest call, but not necessary, probably, since the dependencies are solved taking into account the Python version requirement. If the dependencies are package properly, the risk is reduced.
We can even test all the supported major Python release using a container, actually.